src/view/com/lists/ListItems.tsx

```diff
@@ -124,11 +115,18 @@ export const ListItems = observer(function ListItemsImpl({
   const onPressEditMembership = React.useCallback(
     (profile: AppBskyActorDefs.ProfileViewBasic) => {
       store.shell.openModal({
-        name: 'list-add-remove-user',
+        name: 'user-add-remove-lists',
         subject: profile.did,
         displayName: profile.displayName || profile.handle,
-        onUpdate() {
-          list.refresh()
+        onAdd(listUri: string) {
+          if (listUri === list.uri) {
+            list.cacheAddMember(profile)
+          }
+        },
+        onRemove(listUri: string) {
+          if (listUri === list.uri) {
+            list.cacheRemoveMember(profile)
```

Comment:
Can these optimistic updates happen in the actual add/remove methods? Just feels odd to have them over here separate.

Reply:
That modal uses a different set of ListModels that aren't shared by the one showing the user, unfortunately